ibc/docs/ics02.md

```diff
@@ -0,0 +1,156 @@
+# ICS-02 Client Semantics <!-- omit in toc -->
+
+- [Definitions](#definitions)
+  - ["light client"](#light-client)
+- [Overview](#overview)
+- [Implementation](#implementation)
+  - [Client Manager](#client-manager)
+    - [Lifecycle Management](#lifecycle-management)
+    - [Client Queries](#client-queries)
+- [Types](#types)
+- [Provable Stores](#provable-stores)
+
+## Definitions
+
+### "light client"
+
+In the context of IBC a light client differs from a traditional "light client." An IBC light client is simply a state verification algorithm. It does not sync with the network, it does not download headers. Instead the updates/new headers for a client are provided by an IBC relayer.
+
+## Overview
+
+IBC utilises light clients to verify the correctness of the state of a counterparty chain. This allows for an IBC packet to be committed to in the state of the network on a source chain and then validated through the light client on the counterparty chain.
+
+[ICS-02][ics02] defines the interfaces and types through which the host machine can interact with the light clients it manages. This includes: client creation, client updates and upgrades as well as submitting misbehaviour from the chain the client is tracking. In addition to this, ICS-02 also defines numerous interfaces that are used by the different client implementations in order to carry out the previous actions as well as verify the state of the chain they represent via a proof.
+
+## Implementation
+
+[ICS-02][ics02] is implemented according to the specification. However as the Pocket protocol will utilise [ICS-08][ics08] WASM clients for the improvements to client upgradeability; the implementations of the `ClientState`, `ConsensusState` and other interfaces are specific to a WASM client.
+
+The implementation details are explored below, the code for ICS-02 can be found in [ibc/client](../client/)
+
+### Client Manager
+
+The `ClientManager` is the submodule that governs the light client implementations and implements the [ICS-02][ics02] interface. It is defined in [shared/modules/ibc_client_module.go](../../shared/modules/ibc_client_module.go). The `ClientManager` exposed the following methods:
+
+```go
```

Comment:
I think you should move (if thery're not already there) the interface and comments into the implementation and just link to it. This way there's only one source of truth.